lib/spack/spack/hooks/sbang.py

```diff
@@ -81,8 +81,9 @@ def filter_shebang(path):
     tty.warn("Patched overlong shebang in %s" % path)
 
 
-def filter_shebangs_in_directory(directory):
-    for file in os.listdir(directory):
+def filter_shebangs_in_directory(directory, filenames=None):
+    filenames = filenames or os.listdir(directory)
```

Comment:
This probably results in unexpected behavior if the caller programmatically constructs a list and it ends up being an empty list.  Can you change this to only list if filenames is `None`?